cmd/tools/shared.go

```diff
@@ -0,0 +1,420 @@
+)
+
+// SharedFlags contains common flags used by all tools commands.
+type SharedFlags struct {
+	BinPath          string
+	Config           string
+	DistroLocation   string
+	SkipDepsDownload bool
+	Debug            bool
+	GitProtocol      string
+	OutDir           string
+}
+
+// ToolInfo represents a discovered tool with its version and path.
+type ToolInfo struct {
+	Name       string
+	Version    string
+	BinaryPath string
+	BinaryName string
+}
+
+// OutputFormat represents different shell integration formats.
+type OutputFormat int
+
+// DiscoverTools discovers all available tools and returns their information.
+func DiscoverTools(flags SharedFlags) ([]ToolInfo, error) {
+	// Validate configuration file exists.
+	if _, err := os.Stat(flags.Config); os.IsNotExist(err) {
+		return nil, fmt.Errorf("%w: %s", ErrConfigNotFound, flags.Config)
+	}
+
+	// Initialize paths and clients.
+	outDir := flags.OutDir
+	furyctlPath := flags.Config
+	binPath := flags.BinPath
+
+	if binPath == "" {
+		binPath = path.Join(outDir, ".furyctl", "bin")
+	}
+
+	var err error
+	binPath, err = filepath.Abs(binPath)
+	if err != nil { //nolint:wsl // gofumpt and wsl disagree on formatting
+		return nil, fmt.Errorf("error while getting absolute path for bin folder: %w", err)
+	}
+
+	// Default to https if no protocol specified or invalid.
+	gitProtocol := flags.GitProtocol
+	if gitProtocol == "" {
+		gitProtocol = "https"
+	}
+
+	typedGitProtocol, err := git.NewProtocol(gitProtocol)
+	if err != nil {
+		// Fallback to https on error.
+		typedGitProtocol = git.ProtocolHTTPS
+
+		logrus.Debugf("Invalid git protocol %s, defaulting to https", gitProtocol)
+	}
+
+	// Init packages.
+	execx.Debug = flags.Debug
+	executor := execx.NewStdExecutor()
+	client := netx.NewGoGetterClient()
+
+	// Check if dependencies exist, if not try to download them.
+	needsDownload := false
+	if _, err := os.Stat(binPath); os.IsNotExist(err) {
+		needsDownload = true
+	} else {
+		// Check if directory is empty.
+		entries, err := os.ReadDir(binPath)
+		if err == nil && len(entries) == 0 {
+			needsDownload = true
+		}
+	}
+
+	var distroManifest distroconfig.KFD
+
+	if needsDownload {
+		if flags.SkipDepsDownload {
+			return nil, ErrNoToolsFound
+		}
+
+		// Download distribution and dependencies.
+		var distrodl *dist.Downloader
+		if flags.DistroLocation == "" {
+			distrodl = dist.NewCachingDownloader(client, outDir, typedGitProtocol, "")
+		} else {
+			distrodl = dist.NewDownloader(client, typedGitProtocol, "")
+		}
+
+		// Validate base requirements.
+		depsvl := dependencies.NewValidator(executor, binPath, furyctlPath, false)
+		if err := depsvl.ValidateBaseReqs(); err != nil {
+			return nil, fmt.Errorf("error while validating requirements: %w", err)
+		}
+
+		// Download the distribution (suppress logs for clean output).
+		originalLevel := logrus.GetLevel()
+		logrus.SetLevel(logrus.ErrorLevel)
```

Comment:
why? If I set the log level to DEBUG I want to see debug logs, please don't change it

Reply:
fixd